frontend/device/generic/powerd.lua

```diff
@@ -42,7 +42,8 @@ function BasePowerD:new(o)
 end
 
 function BasePowerD:readyUI()
-    self.UIManager = require("ui/uimanager")
+    UIManager = require("ui/uimanager")
+    self:readyUIHW()
```

Comment:
über nit: you can also simply pass the UIManager reference along, much like what `_setEventHandlers` does ;).

----

Another random idea I had last night would have been to move the full implementation to Generic (while keeping it opt-in, and only actually used on Kobo), with a few knobs via arguments to allow implementations to choose the scheduling frequency and the final delay, if any.

I'm perfectly happy with what you've done here, though, especially since I don't really see us using this on other platforms (I *think* Kindles do the ramp for us, for example). ;).

Reply:
FWIW, that was just a random 3AM idea, I'm perfectly fine with what you've done here ;).